src/frontend/shortlist/src/components/layout/signup_login/__tests__/LoginSignup.test.js

```diff
@@ -0,0 +1,317 @@
+/*
+TODO(All these vue files do not have props that correspond to the API, needs to be added)
+The following tests conclude all major component testing.
+*/
+
+import { describe, it, expect, vi } from "vitest";
+import { mount } from "@vue/test-utils";
+import Signup from "../Signup.vue";
+import Login from "../Login.vue";
+import Logo from "../Logo.vue";
+import LoggedOut from "../LoggedOut.vue";
+import ModalFullScreen from "./../../ModalFullScreen.vue";
+// import axios from "axios";
+import { createTestingPinia } from "@pinia/testing";
+
+describe("Login.vue", () => {
+  // Check if login exists
+  it("Renders with null prop", () => {
+    const componentWrapper = mount(Login, {
+      props: {
+        /*
+        TODO(Add API variables here)
+        */
+      },
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    expect(componentWrapper.exists()).toBe(true);
+  });
+
+  it("Check if the login_components_container exists", async () => {
+    const componentWrapper = mount(Login, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const login_components_container = componentWrapper.find(
+      ".login_components_container"
+    );
+    expect(login_components_container.exists()).toBe(true);
+  });
+
+  it("Check if the loginForm works", async () => {
+    const componentWrapper = mount(Login, {
+      props: {},
+      global: {
+        plugins: [createTestingPinia({ createSpy: vi.fn })],
+      },
+    });
+    const loginForm = componentWrapper.find("#loginForm");
+    expect(loginForm.exists()).toBe(true);
+    await loginForm.trigger("submit.prevent");
```

Comment:
What exactly is this doing here, you triggered a button, but not expect any result? It's not testing anything

Reply:
I am not sure what you meant. I used the codes I saw from the previous test file. If you think I am missing something. Comment here which const or function I am missing? I am trying to test if this button would be trigger-red successfully.

Reply:
It's like you let the test triggered button click, you usually expect a feedback as what will happen, that's the test, but here you triggered the button, then nothing is taking place to check what happened after button press
So you valid the button exist, but didn't check what the button do

Reply:
Yes... Then what feedback? Can you elaborate? What did you do?